tests/integration/models/application-membership.spec.ts

```diff
@@ -0,0 +1,371 @@
+import * as _ from 'lodash';
+import * as m from 'mochainon';
+import {
+	balena,
+	credentials,
+	givenAnApplication,
+	givenLoggedInUser,
+} from '../setup';
+import type * as BalenaSdk from '../../..';
+const { expect } = m.chai;
+import { assertDeepMatchAndLength, timeSuite } from '../../util';
+
+describe('Application Membership Model', function () {
+	timeSuite(before);
+	givenLoggedInUser(before);
+	givenAnApplication(before);
+
+	let ctx: Mocha.Context;
+	before(async function () {
+		ctx = this;
+		const roles = await balena.pine.get({
+			resource: 'application_membership_role',
+			options: { $select: ['id', 'name'] },
+		});
+		this.applicationRoleMap = _.keyBy(roles, 'name');
+		roles.forEach((role) => {
+			expect(role).to.be.an('object');
+			expect(role).to.have.property('id').that.is.a('number');
+		});
+		this.applicationDeveloperRole = this.applicationRoleMap['developer'];
+		this.applicationOperatorRole = this.applicationRoleMap['operator'];
+		this.applicationObserverRole = this.applicationRoleMap['observer'];
+		[
+			this.applicationDeveloperRole,
+			this.applicationOperatorRole,
+			this.applicationObserverRole,
+		].forEach((role) => {
+			expect(role).to.be.an('object');
+		});
+	});
+
+	describe('given a new application', function () {
+		describe('balena.models.application.membership.create()', function () {
+			before(function () {
+				ctx = this;
+			});
+
+			describe('[read operations]', function () {
+				it(`should not be able to add a new member to the application using a wrong role name`, async function () {
+					const promise = balena.models.application.membership.create({
+						application: ctx.application.id,
+						username: credentials.member.username,
+						// @ts-expect-error
+						roleName: 'unknown role',
+					});
+					await expect(promise).to.be.rejected.and.eventually.have.property(
+						'code',
+						'BalenaApplicationMembershipRoleNotFound',
+					);
+				});
+
+				const randomOrdInfo = {
+					id: Math.floor(Date.now() / 1000),
+					app_name: `random_sdk_test_app_name_${Math.floor(Date.now() / 1000)}`,
+					slug: `random_name/random_sdk_test_app_slug_${Math.floor(
+						Date.now() / 1000,
+					)}`,
+				};
+
+				['id', 'app_name', 'slug'].forEach((field) => {
+					it(`should not be able to add a new member when using an not existing application ${field}`, async function () {
+						const promise = balena.models.application.membership.create({
+							application: randomOrdInfo[field],
+							username: credentials.member.username,
+							roleName: 'developer',
+						});
+						await expect(promise).to.be.rejected.and.eventually.have.property(
+							'code',
+							'BalenaApplicationNotFound',
+						);
+					});
+				});
+
+				it(`should not be able to add a new member to a starter application with a given role operator`, async function () {
+					const promise = balena.models.application.membership.create({
+						application: this.application.id,
```

Comment:
I guess that this one was failing and as a result you couldn't use `parallel` right?
```suggestion
						application: ctx.application.id,
```

Reply:
I actually started without using parallel saying "I'll add it later" and then I forgot it. Thank you for pointing it out :)